### PR TITLE
Wrote the UDT's for the car interior in plain English

### PR DESCRIPTION
**Pull request by KiTiKDub**

### CODE REVIEW

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 1]
```diff
@@ -13,75 +13,107 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+    1) Serve Meals
+    2) cook meals
+    3) clean and buss tables
+ */
+
+/*
+Thing 3) Fax Machine/Printer
+5 properties:
+    1) Number of Prints
+    2) Address book
```
Comment (matkatmusic) [inline main.cpp:91]: These are objects, not primitives!   You need to pick things that can be represented with `int float double bool char std::string`
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
One change needed

**Review by matkatmusic — APPROVED** [review]

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 2]
```diff
@@ -12,76 +12,219 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+    pick properties that can be represented with 'int float double bool char std::string'
+    example: 
+        Display:
+            Number of Pixels (int)
+            Amount of Power consumed (milliwatt-hours) (float)
+            Brightness (double)
+            width in cm (int)
+            height in cm (int)
+
+3) move Thing 5-9 to between your Thing 4 and Thing 10.
```
Comment (matkatmusic) [inline main.cpp:80]: you didn't do this.
- [ ] refise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 3]
```diff
@@ -13,75 +13,510 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+     std::string processor = "AMD";
+     // number of cooling gans
+     int coolingFans = 5;
+     //hard drive space (GB)
+     double diskSpace = 1028.0;
+     //Ram storage (GB)
+     int memory = 16;
+
+     struct game
+     {
```
Comment (matkatmusic) [inline main.cpp:174]: Type names need to begin with a capital letter
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -13,75 +13,510 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+    std::string carrier = "AT&T";
+    //Hard Drive Storage (GB)
+    float storage = 64.0f;
+    //Memory
+    float memory = 16.0f;
+    //CPU Speed (GHz)
+    float CPUspeed = 3.6f;
+
+    struct application
+    {
```
Comment (matkatmusic) [inline main.cpp:290]: Type names need to begin with a capital letter
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 5]
```diff
@@ -13,75 +13,511 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+    //Turbine
+    struct Turbine;
+    //Compresser
+    struct Compressor;
+    //Generator
+    struct Generator;
+    //Condenser
+    struct Condenser;
+    //Evaporator
+    struct Evaporator;
```
Comment (matkatmusic) [inline main.cpp:511]: You have forward-declared nested types.
You need to declare instances of your types that already exist.  `#5-#9` are the classes you designed.
This class `PowerPlant` needs to have instances of those classes declared as member variables.
Forward Declaration is covered in Project 5 Part 4 in the Full course.
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -13,75 +13,511 @@ What you create in this project will be used as the basis of Project 5 in the co
...
+
+    //Create energy
+    void createEnergy(Turbine, Compressor, Condenser, Evaporator);
+    //Store Energy
+    void storeEnergy(Generator);
+    //Transfer Energy
+    void transferEnergy(Generator);
```
Comment (matkatmusic) [inline main.cpp:518]: These function parameters should be given names.
- [ ] revise

Also:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course.  If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 5 above)
Comment (KiTiKDub) [reply main.cpp:511]: I don't get any errors when I compile the code. When I made #5-#9, does that not declare those instances within the code already?
